src/llvm-version.h

```diff
@@ -2,53 +2,49 @@
 
 #include <llvm/Config/llvm-config.h>
 
-#if defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 4 && LLVM_VERSION_MINOR >= 0
-#define LLVM40 1
-#endif
-
-#if defined(LLVM40) || (defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 9)
-#define LLVM39 1
-#endif
-
-#if defined(LLVM40) || (defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 8)
-#define LLVM38 1
-#define USE_ORCJIT
-#endif
-
-#if defined(LLVM40) || (defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 7)
-#define LLVM37 1
-
-// We enable ORCJIT only if we have our custom patches
-#ifndef SYSTEM_LLVM
-#define USE_ORCJIT
-#endif
-
-#endif
-
-#if defined(LLVM40) || (defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 6)
-#define LLVM36 1
-#endif
-
-#if defined(LLVM40) || (defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 5)
-#define LLVM35 1
-#endif
-
-#if defined(LLVM40) || (defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR == 3 && LLVM_VERSION_MINOR >= 4)
-#define LLVM34 1
-#define USE_MCJIT
-#endif
-
-#if defined(LLVM_VERSION_MAJOR) && LLVM_VERSION_MAJOR >= 3
-#if defined(LLVM40) || LLVM_VERSION_MINOR >= 3
-#define LLVM33 1
-#endif
-#else
-#error LLVM versions < 3.3 are not supported by Julia
-#endif
-
-#ifdef USE_ORCJIT //temporary, since in some places USE_MCJIT may be used instead of the correct LLVM version test
-#define USE_MCJIT
-#endif
-#ifdef USE_ORCMCJIT //temporary, since in some places USE_MCJIT may be used instead of the correct LLVM version test
-#define USE_MCJIT
+#if defined(LLVM_VERSION_MAJOR) && defined(LLVM_VERSION_MINOR)
+    #if LLVM_VERSION_MAJOR == 4 && LLVM_VERSION_MINOR >= 0
+        #define LLVM_VERSION 40
+    #elif LLVM_VERSION_MAJOR == 3
+        #if LLVM_VERSION_MINOR >= 9
+            #define LLVM_VERSION 39
+        #elif LLVM_VERSION_MINOR == 8
+            #define LLVM_VERSION 38
+        #elif LLVM_VERSION_MINOR == 7
+            #define LLVM_VERSION 37
+        #elif LLVM_VERSION_MINOR == 6
+            #define LLVM_VERSION 36
+        #elif LLVM_VERSION_MINOR == 5
+            #define LLVM_VERSION 35
+        #elif LLVM_VERSION_MINOR == 4
+            #define LLVM_VERSION 34
+        #elif LLVM_VERSION_MINOR == 3
+            #define LLVM_VERSION 33
+        #endif
+    #endif
```

Comment:
Constructing `LLVM_VERSION` can be done more succinctly as
```c
#define CONCAT(a, b) a ## b
#define GETVERS(maj, min) CONCAT(maj, min)
#define LLVM_VERSION GETVERS(LLVM_VERSION_MAJOR, LLVM_VERSION_MINOR)
```
But as @yuyichao noted, this is the kind of thing that one gets stabbed for if put into production.